### PR TITLE
Make sure cvecs are always initialized properly

### PR DESCRIPTION
**Pull request by ajpal**
I got impatient so it hasn't finished running yet, but I think this will fix the soundness issues you've been having with conditionals. Please holler if it doesn't!

### CODE REVIEW

File: tests/bool.rs [hunk 1]
```diff
@@ -358,7 +356,7 @@ mod test {
                 node: 1000000,
             },
         );
-        assert_eq!(can.len(), 12);
-        assert_eq!(cannot.len(), 12);
+        assert!(can.len() > 0);
```
Comment (oflatt) [inline tests/bool.rs:359]: :eye:
why

Comment (oflatt) [reply tests/bool.rs:359]: nvm

**Review by oflatt — APPROVED** [review]
yay Anjali saves the day

**Comment by oflatt** [comment]
Do we still need this?

**Comment by ajpal** [comment]
https://github.com/uwplse/ruler/pull/172 this is a better approach

**Pull request closed without merging**